lambdas/updater/readme.md

```diff
@@ -0,0 +1,282 @@
+        "SigningCertUrl": "https:\/\/sns.eu-west-1.amazonaws.com\/SimpleNotificationService-433026a4050d206028891664da859041.pem",
+        "MessageId": "e91a6f3a-d9dd-5d08-a658-3d16088e9863",
+        "Message": {
+          "name": "S2B_MSIL1C_20170811T012659_N0205_R074_T55UES_20170811T013019",
+          "id": "f32bf1a2-2d71-4a87-a64e-e1a4cd221941",
+          "path": "products\/2017\/8\/11\/S2B_MSIL1C_20170811T012659_N0205_R074_T55UES_20170811T013019",
+          "timestamp": "2017-08-11T01:26:59.027Z",
+          "datatakeIdentifier": "GS2B_20170811T012659_002245_N02.05",
+          "sciHubIngestion": "2017-09-02T09:22:46.926Z",
+          "s3Ingestion": "2017-09-08T17:10:42.312Z",
+          "tiles": [
+            {
+              "path": "tiles\/55\/U\/ES\/2017\/8\/11\/0",
+              "timestamp": "2017-08-11T01:30:19.107Z",
+              "utmZone": 55,
+              "latitudeBand": "U",
+              "gridSquare": "ES",
+              "datastrip": {
+                "id": "S2B_OPER_MSI_L1C_DS_SGS__20170811T064431_S20170811T013019_N02.05",
+                "path": "products\/2017\/8\/11\/S2B_MSIL1C_20170811T012659_N0205_R074_T55UES_20170811T013019\/datastrip\/0"
+              }
+            }
+          ],
+          "datastrips": [
+            {
+              "id": "S2B_OPER_MSI_L1C_DS_SGS__20170811T064431_S20170811T013019_N02.05",
+              "path": "products\/2017\/8\/11\/S2B_MSIL1C_20170811T012659_N0205_R074_T55UES_20170811T013019\/datastrip\/0"
+            }
+          ]
+        },
+        "MessageAttributes": {
+
+        },
+        "Type": "Notification",
+        "UnsubscribeUrl": "https:\/\/sns.eu-west-1.amazonaws.com\/?Action=Unsubscribe&SubscriptionArn=arn:aws:sns:eu-west-1:214830741341:NewSentinel2Product:fbc81a1b-e53f-4cd2-a26b-d70efa534c92",
+        "TopicArn": "arn:aws:sns:eu-west-1:214830741341:NewSentinel2Product",
+        "Subject": "S2B_MSIL1C_20170811T012659_N0205_R074_T55UES_20170811T013019"
+      }
+    }
+  ]
+}
+</pre>
+
+- Run a lambda function each time a new message is received.
+
+  - In python 2.7 (but see below)
+  - Parse the json body of the message
+  - Find and download the metadata of the product and the specific scene (tile) that has been updated, using the name of the product passed in the SNS notification.
+    - Most of this was easy to do using the python libraries in the [Sentinel_S3](https://github.com/developmentseed/sentinel-s3) repository. This repo has the code that is currently used for scraping the whole bucket. I took the functions for parsing single products (`single_metadata`) and called it for each new scene.
```

Comment:
have you checked https://github.com/ungarj/s2reader, it doesn't need gdal C libraries